### ISSUE TITLE
Warn about not using -- after the last option with variables that can be set by the user.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1005]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

set -eu

TMPDIR=${TMPDIR:-/tmp/test}

rm -rf "$TMPDIR"

rm -rf -- "$TMPDIR"

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Shellcheck should warn about `rm -rf "$TMPDIR"` while `rm -rf -- "$TMPDIR"` is correct. `"$TMPDIR"` could be set as an environment variable to anything which includes values like `-foo` which rm will then try to interpret as an option instead of directory or file. To correct this `--` should be passed after the last option to indicate there will be no more options. This also applies to any other variables whose values can be set by the user.
